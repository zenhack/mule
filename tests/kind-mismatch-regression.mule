#XFAIL: expand_path
#
# This is a regression test for a bug where the below failed to type check
# with a kind mismatch error. It *shouldn't* type check, but it *is*
# well-kinded.
4 : (e : { type t a }) -> e.t int
