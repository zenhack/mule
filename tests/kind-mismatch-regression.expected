
Desugared:

	(: (Integer 4)
	 ((e : (exist $6 (record (types ((t $6))) (values ())))) -> ((. e t) int)))
Type error: mismatched type constructors: int and ->
